pkg/databind/pkg/databind/replacer.go

```diff
@@ -286,5 +285,5 @@ func variable(values []data.Map, match []byte, rc replaceConfig) ([]byte, error)
 	}
 
 	// if the value is not found, returns the match itself
-	return match, errors.New("value not found: " + varName)
```

Comment:
What do you think of returning a defined error? (e.g ErrVariableNotFound) And later on, wherever we use this function we can assert the returned error and continue with the execution. It would be nice to have a log entry if is not found (log entry added in epic_logging).

Reply:
The problem with that is that we cannot distinguish between and error (i.e. a discovery variable with a typo) or an expected variable not found (i.e. one used in flex), so the log entry could be confusing.

Reply:
👍  Good catch. I removed it from the non-exported functions but left it in the interface and exported ones.

Reply:
@rogercoll I rebased it so all the changes are in one commit :( 

Changes since last feedback:
* remove the suggested error propagation
* Removed unused ReplaceBytes function